error_check.php

```diff
@@ -0,0 +1,230 @@
+ */
+function check_columns($columns, $table, $operation){
+    // Array containing columns of all tables
+    // Primary key columns are not included
+	$course_question_choice = array('question_id', 'user_id', 'locked', 'position');
+	$courses = array('course_code', 'title', 'department_name');
+	$department = array('name');
+	$dept_question_choice = array('department_id', 'term', 'question_id', 'user_id', 'locked', 'position');
+	$questions = array('answer_type', 'content');
+	$response =array('survey_instance_id', 'question_id', 'answer', 'user_id');
+	$sections = array('course_id', 'term', 'meeting_time', 'room', 'section_code');
+	$survey_instances = array('user_association_id', 'override_token', 'time_window', 'start_time');
+	$surveys =array('name', 'course_id', 'term', 'default_time_window', 'default_start_time');
+	$ta_question_choice = array('section_id', 'term', 'question_id', 'user_id', 'locked', 'position');
+	$user_association =array('user_id', 'course_id', 'section_code');
+	$users =array('utorid', 'type', 'name1', 'photo1');
+
+	$selected_table = NULL;
+	switch($table){
+		case 'course_question_choice':
+			if ($operation!='delete'){
+				$selected_table = $course_question_choice;
+			}
+			break;
+
+		case 'courses':
+			if ($operation!='put'){
+				$selected_table = $courses;
+			}
+			break;
+
+		case 'department':
+			if ($operation!='delete'){
+				$selected_table = $department;
+			}
+			break;
+
+		case 'dept_question_choice':
+			if ($operation!='delete'){
+				$selected_table = $dept_question_choice;
+			}
+			break;
+
+		case 'questions';
+			if (($operation!='put')or($operation!='delete')){
+				$selected_table = $questions;
+			}
+			break;
+
+		case 'response':
+			if (($operation!='put')or($operation!='delete')){
+				$selected_table = $response;
+			}
+			break;
+
+		case 'sections':
+			if ($operation!='put'){
+				$selected_table = $sections;
+			}
+			break;
+
+		case 'survey_instances':
+			if (($operation!='put')or($operation!='delete')){
+				$selected_table = $survey_instances;
+			}
+			break;
+
+		case 'surveys':
+			if (($operation!='put')or($operation!='delete')){
+				$selected_table = $surveys;
+			}
+			break;
+
+		case '$ta_question_choice':
+			if ($operation!='delete'){
+				$selected_table = $ta_question_choice;
+			}
+			break;
+
+		case 'user_association':
+			$selected_table = $user_association;
+			break;
+
+		case 'users':
+			$selected_table = $users;
+			break;
+
+		default:
+			throw new Exception('Table: '. $table . ' does not exist');
+			break;
+	}
+	if(!(isset($selected_table))){
+		throw new Exception("Invalid operation");
+	}
+
+	if ($operation=='post'){
+		if (sizeof($columns)!=sizeof($selected_table)){
+			throw new Exception("Invalid number of columns");
+		}
+	}
+
+	foreach($columns as $column){
+		if (!in_array($column, $selected_table)){
+			throw new Exception('Column: '. $column . ' does not exist');
+		}
+	}
+
+	if ($operation=='put'){
```

Comment:
(So for now you can make this an issue, but don't include it in this version of the code.)